python/marvin/tools/maps.py

```diff
@@ -116,6 +114,8 @@ def __init__(self, input=None, filename=None, mangaid=None, plateifu=None,
 
         self._check_versions(self)
 
+        self.vacs = VACMixIn.get_vacs(self)
```

Comment:
I think it should not be added to Spaxel and Bin for the same reason we don't add the NSA or dapall information there. Once you get a bin or spaxel I think we should remove galaxy-wide metadata.

I'm almost ok with adding it directly to `MarvinToolsClass`, the problem being that `Plate` also inherits from it. Maybe we can manually remove it in `Plate` but it seems a bit like a hack.

Reply:
I only bring this up because Kate explicitly asked in our session if VACS would be available on the Spaxels.  I know her VAC consists of individual spaxel classifications and measurements.   She'll have two files, one of which consists of rows of spaxels not galaxies.    

Yeah, when I made it didn't make much sense for it to be subclassed from `MarvinToolsClass` but I wanted to take advantage of the MMA that it had.  But maybe we can refactor `Plate`.  I've never been entirely happy with it.  But I don't see a huge problem with having vacs attached to `Plate`.  There are only a handful of basic parameters users will need to select rows in their vacs.  We can probably build something into the `Base` that checks for these depending on which `Tool` is being passed.

Reply:
I'd like to see an example of her files and what she'd return before creating a framework for that, but I don't think it's undoable.

I did try to implement a system in which the subclass of `VACMixIn` was able to define in which Marvin classes it wanted to show up but I run into 1001 circular import errors. I may give it another try.

Reply:
I can talk to her about her files again and see if she has produced example versions.  I know she has two kinds of files.  One is a single fits file where each row is an individual spaxel in a galaxy and contains columns like `plateifu`, `bintype`, `template`, `x`, `y` and all of her spaxel classifications.  So this file has millions of rows, spanning all spaxels for all galaxies.   So my guess is each `Spaxel` would return that vac info (PCA classes and params) for that matching row.   

Her other filetype is a directory containing one fits file per galaxy, and it contains basically a new map of her PCA classifications.  So this VAC info would be returned by a `Cube` or `Map`

Reply:
I have added a global `include` attribute that defines in which tool classes the VAC should show up. It always skips the `Plate`. I think that should provide the framework for Kate's VAC. `VACMixIn` now is loaded in the `MarvinToolsClass` and in `SpaxelBase`.